osu.Game/Rulesets/Mods/Mod.cs

```diff
@@ -139,6 +140,30 @@ public virtual Mod DeepClone()
             return result;
         }
 
+        /// <summary>
+        /// Copies all shared mod setting values from <paramref name="source"/> into this instance.
+        /// </summary>
+        /// <param name="source">The mod to copy properties from.</param>
+        public void CopyFromSimilar(Mod source)
```

Comment:
not sure this should be a public method, I don't know that any other usage will ever want to do this - it should definitely not be called `CopyFromSimilar`, maybe `CopySettingsByName` or something